### ISSUE TITLE
cannot use object with `toString()` method as first argument to `fetch()`

### ISSUE THREAD

**Issue by sudo-barun** [microsoft/TypeScript#60379]
### 🔎 Search Terms

fetch, toString

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about fetch and toString.

### ⏯ Playground Link

https://www.typescriptlang.org/play/?#code/GYUwLgxgFgFA3mA9gZTAJwJYDsDmAuAAhgEoCBeAPgIHIowwAHPAemYBtEIBDNqRAZzDUAvsQDcQA

### 💻 Code

```ts
fetch({toString: () => 'http://localhost'});
```

### 🙁 Actual behavior

Errors in code

```
Argument of type '{ toString: () => string; }' is not assignable to parameter of type 'RequestInfo | URL'.
  Type '{ toString: () => string; }' is missing the following properties from type 'URL': hash, host, hostname, href, and 9 more.
```

### 🙂 Expected behavior

No error is expected.

**Comment by MartinJohns** [comment]
This is working as intended. You either need to fully provide a `URL` object, or a `RequestInfo` (which includes `string`). Automatic coercion to `string` when a `toString()` method is present is not supported and would require #35945.
